main/sceneManager.h

```diff
@@ -12,33 +12,34 @@
 #include <string_view>
 #include <vector>
 
+// TODO: change underscore to the end
+
 // This class is creating all necessary objects for simulation.
 class SceneManager
 {
 public:
+    SceneManager();
     bool loadModel(std::string_view objPath);
 
     std::vector<objects::EnergyCollector *> getEnergyCollectors();
-
-    std::vector<objects::TriangleObj *> getModelTriangles()
-    {
-        std::vector<objects::TriangleObj *> v;
-        v.reserve(model_triangles_.size());
-        for (const auto &t : model_triangles_)
-        {
-            v.push_back(t.get());
-        }
-        return v;
-    }
-
+    std::vector<objects::TriangleObj *> getModelTriangles();
     std::vector<objects::TriangleObj *> getReferencePlate();
     const objects::SphereWall &getSphereWall();
+    const generators::PointSource &getPointSource();
 
 private:
     // Disallow copy and assign.
     SceneManager(const SceneManager &) = delete;
     SceneManager &operator=(const SceneManager &) = delete;
-    std::vector<std::unique_ptr<objects::TriangleObj>> model_triangles_;
+
+    void createCollectors();    // assigns energyCollector objects to the right positions. 
+                                // EnergyCollector positions creates two half-circles with the same 
+                                // origin at Vec3(0, 0, 0) that are crossing each other at 
+                                // Vec3(0, 0, kSimulationRadius / 2)
```

Comment:
Aren't they should cross at (0, 0, radius) point?

Reply:
nope. pointSoure needs to be twice as high as energyCollectors.

Reply:
Ok, so the half-circles have kSimulationRadius / 2 radius, which is missing from the comment.